### PR TITLE
BRS-301: Implement International Phone Numbers

### PR DESCRIPTION
**Pull request by Christopher-walsh22**
### Jira Ticket:

BRS-301

### Jira Ticket URL:
[BRS-301](https://github.com/orgs/bcgov/projects/49/views/1?pane=issue&itemId=46688535)

### Description:

Added: ngx-intl-tel-input
Added: google-libphonenumber
Added: ngx-bootstrap

ngx-intl-tel-input is being used to create the dropdown menu in which users can enter internation phone numbers. 

google-libphonenumber will validate the number entered and confirm wether or not that is a valid number in the country selected.

Users have the option to enter: Numbers, spaces, dashes and brackets in the phone number box. This validation will only look at the numbers entered and will not be effected by the other accepted characters.

Update to test to check for validation errors instead of comparing length of entered data to maximum.

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -38,6 +38,7 @@
             ],
             "styles": [
               "node_modules/@digitalspace/bcparks-bootstrap-theme/dist/css/bootstrap-theme.min.css",
+              "./node_modules/intl-tel-input/build/css/intlTelInput.css",
```
Comment (cameronpettit) [inline angular.json:41]: This package doesn't seem to be part of `package.json` - the compiler can't find this file on running the application.

File: src/app/registration/contact-form/contact-form.component.html [hunk 2]
```diff
@@ -103,25 +103,36 @@ <h3>
...
-            name="phone" 
-            placeholder="111-222-3333"
-            maxlength="12"
-            appPhoneFormat 
-            />
-          <div *ngIf="myForm.errors?.['phoneRequired']">
-            <small *ngIf="myForm.errors?.['phoneRequired']" class="text-danger">*Your Phone Number is Required</small>
+        <ngx-intl-tel-input
+            #phoneInput
+            [cssClass]="'custom'"
```
Comment (cameronpettit) [inline src/app/registration/contact-form/contact-form.component.html:108]: Add the `form-control` class to match the field style to other fields on the page.

File: src/app/registration/contact-form/contact-form.component.html [hunk 3]
```diff
@@ -103,25 +103,36 @@ <h3>
...
-            formControlName="phone" 
-            id="phone" 
-            name="phone" 
-            placeholder="111-222-3333"
-            maxlength="12"
-            appPhoneFormat 
-            />
-          <div *ngIf="myForm.errors?.['phoneRequired']">
-            <small *ngIf="myForm.errors?.['phoneRequired']" class="text-danger">*Your Phone Number is Required</small>
+        <ngx-intl-tel-input
```
Comment (cameronpettit) [inline src/app/registration/contact-form/contact-form.component.html:106]: Browser warnings are present:
```
 It looks like you're using ngModel on the same form field as formControlName.
  Support for using the ngModel input property and ngModelChange event with
  reactive form directives has been deprecated in Angular v6 and will be removed
  in a future version of Angular.
```

May need to refactor to clear the deprecation warnings.

File: src/app/registration/contact-form/contact-form.component.ts [hunk 4]
```diff
@@ -116,6 +134,20 @@ export class ContactFormComponent implements OnInit {
...
   }
 
+  updateMaxLength(placeholder: string): void {
+    const placeholderWithoutDialCode = placeholder.slice(this.dialCode.length + 1);
+    this.maxLength = placeholderWithoutDialCode.length;
+    console.log("MAX LENGTH: ", this.maxLength);
+  }
+  
+  onCountryChange(event: any): void{
+    console.log("event: ", event);
```
Comment (cameronpettit) [inline src/app/registration/contact-form/contact-form.component.ts:144]: Remove console logs (typ)

**Review by cameronpettit — CHANGES_REQUESTED** [review]
There are a small number of package issues that should be addressed here.

File: src/app/registration/contact-form/contact-form.component.html (hunk 3 above)
Comment (Christopher-walsh22) [reply src/app/registration/contact-form/contact-form.component.html:106]: Discussed at standup, I will create a new ticket to revise the other fields of the form to only use reactive forms. I have updated the logic for the phone input.

Addressed in: https://github.com/bcgov/parks-reso-public/pull/304/commits/1a599013cd01abb93ec6705be0b8577b355eb442

File: src/app/registration/contact-form/contact-form.component.ts (hunk 4 above)
Comment (Christopher-walsh22) [reply src/app/registration/contact-form/contact-form.component.ts]: Removed this!

Addressed in: https://github.com/bcgov/parks-reso-public/pull/304/commits/1a599013cd01abb93ec6705be0b8577b355eb442

File: src/app/registration/contact-form/contact-form.component.html (hunk 2 above)
Comment (Christopher-walsh22) [reply src/app/registration/contact-form/contact-form.component.html]: I added some custom css to match the styling of the other fields. Adding the form-control class was conflicting with the styles provided from the intl-tel-input library.

Addressed in: https://github.com/bcgov/parks-reso-public/pull/304/commits/1a599013cd01abb93ec6705be0b8577b355eb442

File: angular.json (hunk 1 above)
Comment (Christopher-walsh22) [reply angular.json]: Addressed in: https://github.com/bcgov/parks-reso-public/pull/304/commits/1a599013cd01abb93ec6705be0b8577b355eb442

**Review by cameronpettit — APPROVED** [review]

**Pull request merged**